### DISCUSSION TITLE
How to return 404 error in a custom module

### DISCUSSION THREAD

**Discussion by mohamad-asghari** [dnnsoftware/Dnn.Platform#6068] [Q&A]
How to return 404 error in a custom module? When it returns a 404 error, the 404 page should also be displayed.
For example, the details of a product that has been removed, we want this error to be returned.

**Comment by valadas** [comment]
PortalSettings has a setting called ErrorPage404 which is an integer representing the TabId. Once you get that, you can do something like:

```
Response.Redirect(DotNetNuke.Common.Globals.NavigateURL(tabID), true);
```

Now I am not sure, but I think maybe if there is no custom Error page, that setting may be -1 (or 0) or something... So you may want to try that scenario out. I am guessing that NavigateURL will show the 404 url when passed an invalid tabId, but I did not try.

**Reply by mohamad-asghari** [reply]
Thanks for the answer. I don't want the URL to redirect. Rather, I want the same page with the URL to give the http status code 404 .

**Reply by thienvc** [reply]
Just something like 
`Response.StatusCode = 404`
One more thing, if you want to tell the Search Engine that the content has been deleted (meaning it was once present) then return `Response.StatusCode = 410`
